src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -264,33 +245,21 @@ public void QueueCheckForFilesBeingOpen(ImmutableArray<string> newFileNames)
         public void ProcessQueuedWorkOnUIThread()
             => _openFileTracker?.ProcessQueuedWorkOnUIThread();
 
-        internal void AddProjectToInternalMaps_NoLock(VisualStudioProject project, IVsHierarchy? hierarchy, Guid guid, string projectSystemName)
+        internal void AddProjectToInternalMaps(ProjectSystemProject project, IVsHierarchy? hierarchy, Guid guid, string projectSystemName)
```

Comment:
This is no longer being called under any other lock, so it takes a lock itself. I don't believe this is a particular concern: it means there's a tiny moment in time where a project is added to the CurrentSolution but we won't let have linked up it's hierarchy or GUID; practically I don't imagine that's an issue. The API today where somebody might use that works via VisualStudioProject.GetHierarchy(id) -- since you're passing the ID it's understood that once the project has gone away, you can't get the hierarchy for it, even if you're still holding onto the new project.